rtsf-at-checkout-device-scale/driver/scaleDriver_test.go

```diff
@@ -0,0 +1,401 @@
+// Copyright © 2022 Intel Corporation. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package driver - This package provides a implementation of a ProtocolDriver interface.
+//
+package driver
+
+import (
+	"device-scale/scale"
+	"testing"
+
+	dsModels "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	"github.com/stretchr/testify/require"
+	"go.bug.st/serial.v1/enumerator"
+)
+
+func getDefaultDriverConfig() map[string]string {
+	config := make(map[string]string)
+	config["SimulatorPort"] = "8081"
+	config["ScaleID"] = "123"
+	config["LaneID"] = "123"
+	config["TimeOutMilli"] = "500"
+	return config
+}
+
+func TestScaleDriver_processScaleData(t *testing.T) {
+
+	type fields struct {
+		lc             logger.LoggingClient
+		asyncCh        chan<- *dsModels.AsyncValues
+		scaleDevice    *scaleDevice
+		httpErrors     chan error
+		scaleConnected bool
+		config         map[string]string
+	}
+	type args struct {
+		scaleData     map[string]interface{}
+		deviceResName string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
```

Comment:
Looks like fields is the same for every case.. Can be removed from here and just set one at the top of the test.